test/onyx/log/generative_peer_join.clj

```diff
@@ -485,3 +591,100 @@
     ;; since there are enough peers to handle 2 peers leaving without a task being deallocated the
     ;; job must be able to go on
     (is (>= (apply + (map count (vals (get (:allocations replica) job-1-id)))) 7))))
+
+
+;; Reproduce onyx-test scheduler issue
+(def inner-job-id #uuid "f55c14f0-a847-42eb-81bb-0c0390a88608")
+
+(def inner-job
+  {:workflow [[:a :b] [:b :c] [:c :d] [:d :e]]
+   :catalog [{:onyx/name :a
+              :onyx/plugin :onyx.plugin.core-async/input
+              :onyx/type :input
+              :onyx/min-peers 4
+              :onyx/medium :core.async
+              :onyx/batch-size 20
+              :onyx/doc "Reads segments from a core.async channel"}
+
+             {:onyx/name :b
+              :onyx/fn :mock/fn
+              :onyx/type :function
+              :onyx/batch-size 20}
+
+             {:onyx/name :c
+              :onyx/fn :mock/fn
+              :onyx/type :function
+              :onyx/batch-size 20}
+
+             {:onyx/name :d
+              :onyx/fn :mock/fn
+              :onyx/type :function
+              :onyx/batch-size 20}
+
+             {:onyx/name :e
+              :onyx/plugin :onyx.plugin.core-async/output
+              :onyx/max-peers 1
+              :onyx/type :output
+              :onyx/medium :core.async
+              :onyx/batch-size 20
+              :onyx/doc "Writes segments to a core.async channel"}]
+   :task-scheduler :onyx.task-scheduler/balanced})
+
+(def outer-job-id #uuid "5813d2ec-c486-4428-833d-e8373910ae14")
+
+(def outer-job
+  {:workflow [[:d :e] [:e :f]]
+   :catalog [{:onyx/name :d
+              :onyx/plugin :onyx.plugin.core-async/input
+              :onyx/type :input
+              :onyx/medium :core.async
+              :onyx/batch-size 20
+              :onyx/max-peers 1
+              :onyx/doc "Reads segments from a core.async channel"}
+
+             {:onyx/name :e
+              :onyx/fn :mock/fn
+              :onyx/type :function
+              :onyx/batch-size 20}
+
+             {:onyx/name :f
+              :onyx/plugin :onyx.plugin.core-async/output
+              :onyx/type :output
+              :onyx/max-peers 1
+              :onyx/medium :core.async
+              :onyx/batch-size 20
+              :onyx/doc "Writes segments to a core.async channel"}]
+   :task-scheduler :onyx.task-scheduler/balanced})
+
+(defn running? [peer-counts]
+  (and (not (empty? peer-counts)) 
+       (not (some zero? peer-counts))))
+
+#_(deftest outer-inner-allocations
```

Comment:
Test requires that larger fix to the balanced scheduler that we discussed. We can comment it out when the time comes.

> On 28 Oct 2015, at 4:34 AM, Michael Drogalis <notifications@github.com> wrote:
> 
> In test/onyx/log/generative_peer_join.clj:
> 
> > +              :onyx/batch-size 20}
> > +
> > +             {:onyx/name :f
> > +              :onyx/plugin :onyx.plugin.core-async/output
> > +              :onyx/type :output
> > +              :onyx/max-peers 1
> > +              :onyx/medium :core.async
> > +              :onyx/batch-size 20
> > +              :onyx/doc "Writes segments to a core.async channel"}]
> > +   :task-scheduler :onyx.task-scheduler/balanced})
> > +
> > +(defn running? [peer-counts]
> > +  (and (not (empty? peer-counts)) 
> > +       (not (some zero? peer-counts))))
> > +
> > +#_(deftest outer-inner-allocations
> Should this be commented out, or is it intended to be live?
> 
> —
> Reply to this email directly or view it on GitHub.
>